### PR TITLE
Fix API Explorer usage with OData.

### PR DESCRIPTION
**Pull request by midgleyc**
### Issues

*This pull request fixes issue #1750.*

### Description

*This PR changes ODataInputFormatter and ODataOutputFormatter to override GetSupportedContentTypes to not throw an exception when SupportedMediaTypes is empty.*

This lead to a problem when using Swagger with a controller with both OData and non-OData routes, as seen in https://github.com/OData/WebApi/issues/1177#issuecomment-353738261.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

I don't know what the "one-click build and test script" is, but I ran the newly added unit tests and they pass with this change, and fail without.

### CODE REVIEW

**Comment by midgleyc** [comment]
Fixes #1750.

File: test/UnitTest/Microsoft.Test.AspNet.OData/Formatter/ODataMediaTypeFormattersTests.cs [hunk 1]
```diff
@@ -673,6 +673,50 @@ public void TestCreate_DollarFormat_Error(string dollarFormatValue, string expec
...
+            foreach (var formatter in formatters)
+            {
+                formatter.SupportedMediaTypes.Clear();
+            }
+
+            // Assert
+            foreach (var formatter in formatters)
+            {
+                var exception = Record.Exception(() => formatter.GetSupportedContentTypes("application/json", typeof(string)));
+                Assert.Null(exception);
```
Comment (KenitoInc) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Formatter/ODataMediaTypeFormattersTests.cs:694]: Add another `Assert` for the expected value when calling `GetSupportedContentTypes`

File: test/UnitTest/Microsoft.Test.AspNet.OData/Formatter/ODataMediaTypeFormattersTests.cs [hunk 2]
```diff
@@ -673,6 +673,50 @@ public void TestCreate_DollarFormat_Error(string dollarFormatValue, string expec
...
```
Comment (KenitoInc) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Formatter/ODataMediaTypeFormattersTests.cs:715]: Add another `Assert` for the expected value when calling `GetSupportedContentTypes`

**Review by KenitoInc — CHANGES_REQUESTED** [review]

**Review by KenitoInc — APPROVED** [review]

**Comment by midgleyc** [comment]
I pointed this at `feature/netcore` because reading the descriptions I thought that was the right branch, but it was last updated in 2018. Should I have set it to merge into `master` instead? It rebases cleanly on master.

**Comment by KenitoInc** [comment]
> I pointed this at `feature/netcore` because reading the descriptions I thought that was the right branch, but it was last updated in 2018. Should I have set it to merge into `master` instead? It rebases cleanly on master.

Point it to master

**Review by gathogojr — APPROVED** [review]
LGTM

**Comment by gathogojr** [comment]
@midgleyc Can you follow the instructions [here](https://github.com/OData/WebApi/pull/2494/checks?check_run_id=2836870786) to update the baseline? That should fix the failing tests. If you're unable to please let me know so I can clone the branch and update

**Review by habbes — APPROVED** [review]

**Pull request merged**